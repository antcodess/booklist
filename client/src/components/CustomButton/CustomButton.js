import React from 'react';

import './CustomButton.scss';

const CustomButton = ({ children, isGoogleSignIn, isRegularSignIn, ...otherProps }) => {
  return (
    <button 
    className={`custom-button ${isGoogleSignIn ? 'google-sign-in' : ''}${isRegularSignIn ? 'regular-sign-in' : ''}`} 
    {...otherProps}>{ children }</button>
  );
};

export default CustomButton;