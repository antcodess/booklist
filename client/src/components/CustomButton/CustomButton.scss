.regular-sign-in {
  margin-bottom: 5px;
  &:hover {
    background-color: black;
    opacity: 0.9;
  }
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    opacity: 0.9;
  }
}

@media only screen and (min-width: 950px) {
  .regular-sign-in {
    width: 40%;
  }
}

@media only screen and (min-width: 1100px) {
  .regular-sign-in, .google-sign-in {
    width: 49%;
  }
}