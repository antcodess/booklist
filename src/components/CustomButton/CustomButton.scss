.regular-sign-in {
  margin-bottom: 5px;
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    background-color: black;
  }
}