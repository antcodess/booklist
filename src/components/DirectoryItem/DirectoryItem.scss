.directory-item {
  position: relative;
  cursor: pointer;

  .directory-item-image {

    .image {
      border-radius: 10px;
      max-width: 100%;
      height: auto;
    }
  }

  h4 {
    margin-top: 10px;
    margin-bottom: 2px;
  }
  
  .custom-button {
    display: none;
    width: 90%;
    top: 58%;
    right: 5%;
    position: absolute;
  }

  &:hover {
    .custom-button {
      display: flex;
    }
    
    .image {
      opacity: 0.8;
    }
  }
}