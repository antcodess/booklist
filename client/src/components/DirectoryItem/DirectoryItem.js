import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';

import { addItem } from '../../redux/cart/cart.actions';

import CustomButton from '../CustomButton/CustomButton';

import './DirectoryItem.scss';

const DirectoryItem = ({ item, history, addItem }) => {
  const { title, imageUrl, price } = item;
  return (
    <div className='directory-item'>
      <div className='directory-item-image' onClick={() => history.push(`/${item.routeName}`)}>
        <img className='image' src={ imageUrl } alt='products' />
      </div>
      <div className='directory-meta' onClick={() => history.push(`/${item.routeName}`)}>
        <h4>{ title }</h4>
        <span>${ price }</span>
      </div>
      <CustomButton onClick={() => addItem(item)}>Add To Cart</CustomButton>
    </div>
  );
};

const mapDispatchToProps = dispatch => ({
  addItem: item => dispatch(addItem(item))
});

export default connect(null, mapDispatchToProps)(withRouter(DirectoryItem));