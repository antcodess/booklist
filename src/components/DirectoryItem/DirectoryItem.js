import React from 'react';


import './DirectoryItem.scss';

const DirectoryItem = ({ photo, name, price }) => {
  return (
    <div className='directory-item'>
      <div className='directory-item-image'>
        <img src={ photo } alt='products' />
      </div>
      <h4>{ name }</h4>
      <span>{ price }</span>
    </div>
  );
};

export default DirectoryItem;