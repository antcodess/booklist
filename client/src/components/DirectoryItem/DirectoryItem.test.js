import React from 'react';
import { shallow } from 'enzyme';
import { DirectoryItem } from './DirectoryItem';
import CustomButton from '../CustomButton/CustomButton';
import { mapDispatchToProps } from './DirectoryItem';

describe('DirectoryItem Component', () => {
  let wrapper;
  let mockItem;
  let mockAddItem;
  let mockHistory;

  const mockAddItemAction = {
    type: 'ADD_ITEM', 
    payload: {
      title: 'Item 1', 
      imageUrl: 'example.com', 
      quantity: 1, 
      price: 20, 
      routeName: '/products/item-1'
    }
  }
  
  beforeEach(() => {

    mockItem = {
      title: 'Item 1', 
      imageUrl: 'example.com', 
      quantity: 1, 
      price: 20, 
      routeName: '/products/item-1'
    };

    mockAddItem = jest.fn();
    mockHistory = { push: jest.fn() };

    const mockProps = {
      item: mockItem, 
      addItem: mockAddItem, 
      history: mockHistory
    };

    wrapper = shallow(<DirectoryItem { ...mockProps } />);
  });

  it('should render DirectoryItem Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked for directory-item-image div', () => {
    wrapper.find('.directory-item-image').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call history.push when clicked for directory-meta div', () => {
    wrapper.find('.directory-meta').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should call addItem when clicked', () => {
    wrapper.find(CustomButton).simulate('click');
    expect(mockAddItem).toHaveBeenCalled();
  });

  it('should dispatch addItem action when clicked', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).addItem(mockItem);
    expect(dispatch.mock.calls[0][0]).toEqual(mockAddItemAction);
  });
});