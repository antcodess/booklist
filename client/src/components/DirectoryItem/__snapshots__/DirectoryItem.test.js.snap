// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectoryItem Component should render DirectoryItem Component 1`] = `ShallowWrapper {}`;
